extras/editor/emacs/lambda-buffers-mode.el

```diff
@@ -0,0 +1,16 @@
+;;; lambda-buffers-mode.el --- SUMMARY -*- lexical-binding: t -*-
+
+(require 'haskell-mode)
+
+(define-derived-mode lambda-buffers-mode haskell-mode "lambda-buffers"
+  "haskell-mode pretending to be lambda-buffers-mode"
+  (haskell-indent-mode)
+  (font-lock-add-keywords
+   nil
+   '(("\\<\\(record\\|sum\\|derive\\|opaque\\|prod\\)\\>" . 'font-lock-keyword-face)))
```

Comment:
We also have `instance` keyword, should we add it here?

Instance definitions are a declaration of 'this semantics is supported by the implementation' so for example, we have a `Maybe` opaque type in Prelude schema and say `instance Eq (Maybe a) :- Eq a`. The syntax is Prolog syntax :D because it was easier to parse